### ISSUE TITLE
Crashes at startup when using "use-app-thinning"

### ISSUE THREAD

**Issue by AndreaVitale** [hansemannn/titanium-google-maps#83]
Hi @hansemannn, first of all thank for your amazing work!
I'm trying to use this module with Alloy on an app with Ti SDK 5.5.1.GA, but it crashes at startup with this stacktrace:

```
[DEBUG] Mar 14 08:59:56  google-maps[12602] <Error>: CGContextSetFillColorWithColor: invalid context 0x0. If you want to see the backtrace, please set CG_CONTEXT_SHOW_BACKTRACE environmental variable.
[DEBUG] Mar 14 08:59:56  google-maps[12602] <Error>: CGContextSetStrokeColorWithColor: invalid context 0x0. If you want to see the backtrace, please set CG_CONTEXT_SHOW_BACKTRACE environmental variable.
[DEBUG] Mar 14 08:59:56  google-maps[12602] <Error>: CGContextGetCompositeOperation: invalid context 0x0. If you want to see the backtrace, please set CG_CONTEXT_SHOW_BACKTRACE environmental variable.
[DEBUG] Mar 14 08:59:56  google-maps[12602] <Error>: CGContextSetCompositeOperation: invalid context 0x0. If you want to see the backtrace, please set CG_CONTEXT_SHOW_BACKTRACE environmental variable.
[DEBUG] Mar 14 08:59:56  google-maps[12602] <Error>: CGContextFillRects: invalid context 0x0. If you want to see the backtrace, please set CG_CONTEXT_SHOW_BACKTRACE environmental variable.
[DEBUG] Mar 14 08:59:56  google-maps[12602] <Error>: CGContextSetCompositeOperation: invalid context 0x0. If you want to see the backtrace, please set CG_CONTEXT_SHOW_BACKTRACE environmental variable.
[DEBUG] Mar 14 08:59:56  google-maps[12602] <Error>: CGContextSetFillColorWithColor: invalid context 0x0. If you want to see the backtrace, please set CG_CONTEXT_SHOW_BACKTRACE environmental variable.
[DEBUG] Mar 14 08:59:56  google-maps[12602] <Error>: CGContextSetStrokeColorWithColor: invalid context 0x0. If you want to see the backtrace, please set CG_CONTEXT_SHOW_BACKTRACE environmental variable.
[DEBUG] Mar 14 08:59:56  google-maps[12602] <Error>: CGContextGetBlendMode: invalid context 0x0. If you want to see the backtrace, please set CG_CONTEXT_SHOW_BACKTRACE environmental variable.
[DEBUG] Mar 14 08:59:56  google-maps[12602] <Error>: CGContextSetBlendMode: invalid context 0x0. If you want to see the backtrace, please set CG_CONTEXT_SHOW_BACKTRACE environmental variable.
[DEBUG] Mar 14 08:59:56  google-maps[12602] <Error>: CGContextFillRects: invalid context 0x0. If you want to see the backtrace, please set CG_CONTEXT_SHOW_BACKTRACE environmental variable.
[DEBUG] Mar 14 08:59:56  google-maps[12602] <Error>: CGContextSetBlendMode: invalid context 0x0. If you want to see the backtrace, please set CG_CONTEXT_SHOW_BACKTRACE environmental variable.
[DEBUG] Mar 14 08:59:56  google-maps[12602] <Error>: CGContextSetFillColorWithColor: invalid context 0x0. If you want to see the backtrace, please set CG_CONTEXT_SHOW_BACKTRACE environmental variable.
[DEBUG] Mar 14 08:59:56  google-maps[12602] <Error>: CGContextSetStrokeColorWithColor: invalid context 0x0. If you want to see the backtrace, please set CG_CONTEXT_SHOW_BACKTRACE environmental variable.
[DEBUG] Mar 14 08:59:56  google-maps[12602] <Error>: CGContextGetBlendMode: invalid context 0x0. If you want to see the backtrace, please set CG_CONTEXT_SHOW_BACKTRACE environmental variable.
[DEBUG] Mar 14 08:59:56  google-maps[12602] <Error>: CGContextSetBlendMode: invalid context 0x0. If you want to see the backtrace, please set CG_CONTEXT_SHOW_BACKTRACE environmental variable.
[DEBUG] Mar 14 08:59:56  google-maps[12602] <Error>: CGContextFillRects: invalid context 0x0. If you want to see the backtrace, please set CG_CONTEXT_SHOW_BACKTRACE environmental variable.
[DEBUG] Mar 14 08:59:56  google-maps[12602] <Error>: CGContextSetBlendMode: invalid context 0x0. If you want to see the backtrace, please set CG_CONTEXT_SHOW_BACKTRACE environmental variable.
[INFO]  appcelerator: Appcelerator (San Jose, CA)
[INFO]  apple: Apple (Cupertino, CA)
[INFO]  facebook: Facebook (Menlo Park, CA)
[INFO]  microsoft: Microsoft (Redmond, WA)
[INFO]  2017-03-14 08:59:56.886 google-maps[12602:44354739] ((null)) was false: Cannot find image GMSSprites-0-2x
[INFO]  2017-03-14 08:59:56.887 google-maps[12602:44354739] ((null)) was false: Image data can't be NULL.
[INFO]  2017-03-14 08:59:56.887 google-maps[12602:44354739] ((null)) was false: Cannot find image GMSNavSprites-0-2x
[INFO]  2017-03-14 08:59:56.887 google-maps[12602:44354739] ((null)) was false: Image data can't be NULL.
[INFO]  2017-03-14 08:59:56.887 google-maps[12602:44354739] ((null)) was false: Cannot find image GMSNavNightModeSprites-0-2x
[INFO]  2017-03-14 08:59:56.888 google-maps[12602:44354739] ((null)) was false: Image data can't be NULL.
[INFO]  2017-03-14 08:59:56.888 google-maps[12602:44354739] ((null)) was false: Cannot find image polyline_colors_texture
[INFO]  2017-03-14 08:59:56.889 google-maps[12602:44354739] ((null)) was false: Image data can't be NULL.
[INFO]  2017-03-14 08:59:56.889 google-maps[12602:44354739] ((null)) was false: Invalid texture state for routesTextureState.
[INFO]  2017-03-14 08:59:56.889 google-maps[12602:44354739] ((null)) was false: Cannot find image polyline_colors_texture_dim
[INFO]  2017-03-14 08:59:56.889 google-maps[12602:44354739] ((null)) was false: Image data can't be NULL.
[INFO]  2017-03-14 08:59:56.890 google-maps[12602:44354739] ((null)) was false: Invalid texture state for routesDimTextureState.
```

I have a reference to the module inside `alloy.js`:

```
Alloy.Globals.GoogleMaps = require('ti.googlemaps');
Alloy.Globals.GoogleMaps.setAPIKey('API_KEY');
```

and inside my `index.js` I have:

```
var companies = {
    appcelerator: {
        title: "Appcelerator",
        city: "San Jose, CA",
        latitude: 37.368122,
        longitude: -121.913653
    },
    apple: {
        title: "Apple",
        city: "Cupertino, CA",
        latitude: 37.331711,
        longitude: -122.030184
    },
    facebook: {
        title: "Facebook",
        city: "Menlo Park, CA",
        latitude: 37.4748624,
        longitude: -122.1490817
    },
    microsoft: {
        title: "Microsoft",
        city: "Redmond, WA",
        latitude: 47.641959,
        longitude: -122.130588
    }
};

$.Map.region = {
    latitude: companies.appcelerator.latitude,
    longitude: companies.appcelerator.longitude,
    zoom: 10
};

addMarkers(companies);

$.index.open();

function addMarkers(companies) {
    for (var key in companies)
        if (companies.hasOwnProperty(key)) {
            var company = companies[key];

            var annotation = Alloy.Globals.GoogleMaps.createAnnotation({
                latitude: company.latitude,
                longitude: company.longitude,
                title: company.title,
                subtitle: company.city
            });

            $.Map.addAnnotation(annotation);

            Ti.API.info(key + ": " + company.title + " (" + company.city + ")");
        }
}
```

How can I solve? 
Thank you.

**Comment by hansemannn** [comment]
- Did you set a valid API key that matches with your bundle ID?
- Did you ensure the bundle is copied to the app ([check](https://www.b4x.com/android/forum/threads/googlemaps-example.46637/))?

We has this report many times now, it was always one of the above issues on the devs side.

**Comment by AndreaVitale** [comment]
Previously I had an unrestricted API Key. Now I create an iOS App API Key with the same bundle id but I got the same error.

**Comment by hansemannn** [comment]
Mhhh..the error looks like something different. Would you be able to provide a sample project or tiapp.xml?

**Comment by AndreaVitale** [comment]
Here is the sample app. I attach also the screenshot from google developer console.
[google-maps.zip](https://github.com/hansemannn/ti.googlemaps/files/840745/google-maps.zip)

<img width="606" alt="schermata 2017-03-14 alle 10 04 19" src="https://cloud.githubusercontent.com/assets/2265079/23893731/e693bd96-089f-11e7-9c2a-b6996637f974.png">
<img width="311" alt="schermata 2017-03-14 alle 10 05 23" src="https://cloud.githubusercontent.com/assets/2265079/23893732/e6b4f5b0-089f-11e7-8d31-191d70b3eb2d.png">

**Comment by hansemannn** [comment]
That's the bad boy:
```xml
<use-app-thinning>true</use-app-thinning>
```
Seems like there is an SDK issue with app-thinning & modules bundles. I'll file a ticket for that. Removing that lines fixes it for me.

**Comment by Topener** [comment]
Can confirm this also fixes it for me!

**Comment by AndreaVitale** [comment]
I confirm that it works also for me, thank you :)

**Comment by hansemannn** [comment]
Ticket filed: https://jira.appcelerator.org/browse/TIMOB-24484

**Comment by hansemannn** [comment]
Core-PR available [here](https://github.com/appcelerator/titanium_mobile/pull/8886), try it out!

**Comment by hansemannn** [comment]
Will be resolved in 6.1.0. Grab the PR from above if you'r experiencing the issue. I'll keep this ticket open for now, so other people can refer to the fix.

**Comment by hansemannn** [comment]
PR was merged, available in 6.1.0.
